contracts/BosonRouter.sol

```diff
@@ -883,19 +883,29 @@ contract BosonRouter is
     ) internal {
         address tokenWrapper = ITokenRegistry(tokenRegistry)
             .getTokenWrapperAddress(_token);
-        require(tokenWrapper != address(0), "UNSUPPORTED_TOKEN");
-
-        //The BosonToken contract conforms to this spec, so it will be callable this way
-        //if it's address is mapped to itself in the TokenRegistry
-        ITokenWrapper(tokenWrapper).permit(
-            _tokenOwner,
-            _spender,
-            _value,
-            _deadline,
-            _v,
-            _r,
-            _s
-        );
+      
+        if(tokenWrapper != address(0)) { //tokens like DAI don't conform to EIP-2612, so a token wrapper is needed
```

Comment:
Since the method on the `ITokenWrapper.permit` and the `IERC20WIthPermit.permit` have the same signature and are interchangeable for the purpose of this call, all we need to do here is alternate based on the expression. This if/else block could be replaced by:

```solidity
bool isWrapped = ( tokenWrapper != address(0) );
IERC20WithPermit( isWrapped ? tokenWrapper : _token );
```

Reply:
Modified.